src/xcode/ENA/ENA/Source/Services/CCLService/CCLService.swift

```diff
@@ -335,14 +325,38 @@ class CCLService: CCLServable {
 		}
 	}
 
-	private func updateJsonFunctions(
-		_ configurations: [CCLConfiguration]
+	private func replaceCCLConfigurations(
+		with newCCLConfigurations: [CCLConfiguration]
 	) {
-		configurations.forEach { [weak self] configuration in
-			configuration.functionDescriptors.forEach { jsonFunctionDescriptor in
-				self?.jsonFunctions.registerFunction(jsonFunctionDescriptor: jsonFunctionDescriptor)
+		/// Reset registered functions by creating a new instance
+		jsonFunctions = JsonFunctions()
+
+		for configuration in newCCLConfigurations {
+			registerJsonFunctions(from: configuration)
+		}
+
+		var registeredConfigurations = newCCLConfigurations
+
+		/// Register functions from the default configurations as well, in case the default configurations contain (new) configurations not contained in the cached/fetched configurations
+		if let defaultConfigurations = cclConfigurationResource.defaultModel()?.cclConfigurations {
```

Comment:
I think the CCLService should not have a dependency to CCLConfigurationResource.
Instead it should load the default on its own.